db/migrate/20231025131430_add_author_ref_to_post.rb

```diff
@@ -0,0 +1,7 @@
+class AddAuthorRefToPost < ActiveRecord::Migration[7.1]
+  def change
+    add_column :posts, :author_id, :string
+    add_reference :posts, :author_id, null: false, foreign_key: {to_table: :users}
+    add_index :posts, :author_id
+  end
```

Comment:
- [ ] @Wineshuga Great work adding references for users in the posts table, however in this migration file you are adding  author_id before using it as a reference, which results in the table schema shown below, kindly consider using the suggestion below to fix the issue.
![image](https://github.com/Wineshuga/pen-it/assets/61408860/b0d48c97-1dc7-4aa9-8c14-4f045c5764f8)
```suggestion
  def change
    add_reference :posts, :author, null: false, foreign_key: {to_table: :users}
  end
```